common/src/main/java/com/microsoft/identity/common/internal/ui/webview/switchbrowser/SwitchBrowserUriHelper.kt

```diff
@@ -171,6 +172,77 @@ object SwitchBrowserUriHelper {
         Logger.info(methodTag, "States match.")
     }
 
+    /**
+     * Validates that the provided authority is a valid Azure Active Directory (AAD) authority.
+     *
+     * This method performs the following validations:
+     * 1. Checks if the authority parameter is not null or empty
+     * 2. Ensures AzureActiveDirectory is initialized (performs cloud discovery if needed)
+     * 3. Verifies the authority exists in the list of known AAD hosts
+     *
+     * @param authority The authority URL to validate (e.g., "login.microsoftonline.com")
+     *
+     * @throws ClientException with [ClientException.MISSING_PARAMETER] if authority is null or empty
+     * @throws ClientException with [ClientException.UNKNOWN_AUTHORITY] if authority is not a valid AAD authority
+     *
+     * @see AzureActiveDirectory.getHosts
+     * @see AzureActiveDirectory.performCloudDiscovery
+     */
+    fun validateAadAuthority(authority: String?) {
```

Comment:
I think this method could easily move into the Authority class to become a generic authority related method that anyone can invoke (not just switch browser)